project/src/components/login/login.tsx

```diff
@@ -1,29 +1,43 @@
 import Sprite from '../common/sprite';
 import Logo from '../common/logo';
-import { FormEvent, useRef } from 'react';
+import { FormEvent, memo, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { loginAction } from '../../store/api-actions';
-import { AppRoutes } from '../../const';
 import { useHistory } from 'react-router';
-import { State } from '../../types/reducer';
-import { Link } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import { AppRoutes, AuthorizationStatus } from '../../const';
+import { statusOfAuth } from '../../store/reducer/user/user-selectors';
+import RandomCity from './random-city';
+import LoginInput from './login-input';
+import PasswordInput from './password-input';
+
+const WRONG_PASSWORD = 'Пароль должен состоять минимум из одной буквы и одной цифры';
 
 function LogIn(): JSX.Element {
   const history = useHistory();
   const dispatch = useDispatch();
-  const currentCity = useSelector(({ city }: State) => city);
-  const loginRef = useRef<HTMLInputElement | null>(null);
-  const passwordRef = useRef<HTMLInputElement | null>(null);
+  const authStatus = useSelector(statusOfAuth);
+
+  if (authStatus === AuthorizationStatus.AUTH) {
+    history.push(AppRoutes.Main);
+  }
 
+  const [password, setPassword] = useState('');
+  const [email, setEmail] = useState('');
 
   const handleSubmit = (evt: FormEvent<HTMLFormElement>) => {
     evt.preventDefault();
-
-    if (loginRef.current !== null && passwordRef.current !== null) {
+    const testPassword = /([A-zА-я]{1}[0-9]{1})|([0-9]{1}[A-zА-я]{1})/;
+    if (testPassword.test(password)) {
```

Comment:
```suggestion
    const testPassword = /([A-zА-я]{1}[0-9]{1})|([0-9]{1}[A-zА-я]{1})/;
    
    if (testPassword.test(password)) {
```

ну и паттерн вынеси в отдельную константу, зачем тебе на каждый вызов коллбека создавать этот паттерн?